test/examples/uct_hello_world.c

```diff
@@ -200,34 +220,40 @@ ucs_status_t do_am_zcopy(iface_info_t *if_info, uct_ep_h ep, uint8_t id,
     return status;
 }
 static void print_strings(const char *label, const char *local_str,
-                          const char *remote_str)
+                          const char *remote_str, size_t length)
 {
-    fprintf(stdout, "\n\n----- UCT TEST SUCCESS ----\n\n");
-    fprintf(stdout, "[%s] %s sent %s", label, local_str, remote_str);
-    fprintf(stdout, "\n\n---------------------------\n");
-    fflush(stdout);
+    char *str = calloc(1, length);
+    if (str != NULL) {
+        mem_type_allocators[test_mem_type].memcpy(str, remote_str, length);
+        fprintf(stdout, "\n\n----- UCT TEST SUCCESS ----\n\n");
+        fprintf(stdout, "[%s] %s sent %s", label, local_str, str);
+        fprintf(stdout, "\n\n---------------------------\n");
+        fflush(stdout);
+        free(str);
+    }
 }
 
 /* Callback to handle receive active message */
 static ucs_status_t hello_world(void *arg, void *data, size_t length, unsigned flags)
 {
     recv_desc_t *rdesc;
     func_am_t func_am_type = *(func_am_t *)arg;
-    print_strings("callback", func_am_t_str(func_am_type), data);
+    print_strings("callback", func_am_t_str(func_am_type), data, length);
 
     if (flags & UCT_CB_PARAM_FLAG_DESC) {
         rdesc = (recv_desc_t *)data - 1;
         /* Hold descriptor to release later and return UCS_INPROGRESS */
-        rdesc->is_uct_desc = 1;
+        set_uct_desc_flag(rdesc, 1);
```

Comment:
`rdesc` should always be a host memory. With AM,  data arrives to preposted HOST buffers